### PR TITLE
[master-next] Adjust to upstream llvm change SectionRef::getName()

### PR DESCRIPTION
**Pull request by aschwaighofer**

### CODE REVIEW

File: lib/IRGen/IRGen.cpp [hunk 1]
```diff
@@ -384,7 +384,12 @@ static bool needsRecompile(StringRef OutputFilename, ArrayRef<uint8_t> HashData,
   // Search for the section which holds the hash.
   for (auto &Section : ObjectFile->sections()) {
     StringRef SectionName;
-    Section.getName(SectionName);
+    if (auto nameOrErr = Section.getName())
+      SectionName = *nameOrErr;
+    else {
```
Comment (compnerd) [inline lib/IRGen/IRGen.cpp]: Could you please use braces on both sides?  Its harder to read if only one side uses braces.

File: tools/lldb-moduleimport-test/lldb-moduleimport-test.cpp [hunk 2]
```diff
@@ -168,7 +168,10 @@ collectASTModules(llvm::cl::list<std::string> &InputNames,
 
     for (auto &Section : Obj->sections()) {
       llvm::StringRef Name;
-      Section.getName(Name);
+      auto nameOrErr = Section.getName();
+      if (nameOrErr.takeError())
```
Comment (compnerd) [inline tools/lldb-moduleimport-test/lldb-moduleimport-test.cpp]: I think you should use the bool conversion instead:

```
if (!nameOrErr) {
  reportError(nameOrErr.takeError());
  continue;
}
```

**Comment by xiaobai** [comment]
I've got a similar patch here: https://github.com/apple/swift/pull/26657

I don't mind which one goes in. Glad to know others are working on this! :)

**Comment by aschwaighofer** [comment]
@xiaobai Oh I missed your PR.

**Pull request closed without merging**